collectors/test-results/jira-xray/src/main/java/com/capitalone/dashboard/core/json/TestStepJsonParser.java

```diff
@@ -0,0 +1,55 @@
+package com.capitalone.dashboard.core.json;
+
+import com.atlassian.jira.rest.client.internal.json.JsonObjectParser;
+import com.atlassian.jira.rest.client.internal.json.JsonParseUtil;
+import com.capitalone.dashboard.api.domain.TestStep;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+
+import java.net.URI;
+
+/*This class will parse the JSON for a Test Step
+ */
+
+public class TestStepJsonParser implements JsonObjectParser<TestStep> {
+
+    private static final RendereableItemJsonParser RENDEREABLE_JSON_PARSER =new RendereableItemJsonParser();
+
+    private final static String KEY_ID="id";
+    private final static String KEY_INDEX="index";
+    private final static String KEY_STEP="step";
+    private final static String KEY_DATA="data";
+    private final static String KEY_RESULT="result";
+    private final static String KEY_STATUS="status";
+
+    public TestStep parse(JSONObject jsonObject) throws JSONException {
+        jsonObject.put("self",""); // TODO: ADD URI.
+        URI selfUri = JsonParseUtil.getSelfUri(jsonObject);
+        String key =" THERE IS NO KEY FOR TEST RUN AT X-RAY DIRECT REST API"; // TODO: GET THE ISSUE KEY
```

Comment:
@kumarvarun1252 key value is assigned as THERE IS NO KEY FOR TEST RUN AT X-RAY DIRECT REST API and never re-assigned. is this value passed to create TestRun

Reply:
Same here, optional field doesn't affect parsing TestRun. Added TODO